pkgs/development/ocaml-modules/elpi/default.nix

```diff
@@ -7,41 +8,41 @@
 , re, perl, ncurses
 , ppxlib, ppx_deriving
 , ppxlib_0_15, ppx_deriving_0_15
-, coqPackages
 , version ? if lib.versionAtLeast ocaml.version "4.08" then "1.18.1"
-    else if lib.versionAtLeast ocaml.version "4.07" then "1.15.2" else "1.14.1"
+    else "1.15.2"
 }:
 
 let p5 = camlp5; in
 let camlp5 = p5.override { legacy = true; }; in
 
-let fetched = coqPackages.metaFetch ({
```

Comment:
@vbgl removing this will break the ability to conveniently overlay ocamlPackages.elpi in various CIs to anything else than released versions, which might be pretty unconvenient. Although I'm not sure how often we use that feature.
Cc @CohenCyril @gares ?

Reply:
Right. So my question is: can we use `metaFetch` to download the released artifacts when available?

Reply:
I use that feature via the coq nix toolbox.
But I also need elpi to have a version number (for conditional compilation).

I hope both requirements are not in conflict.

Reply:
> can we use `metaFetch` to download the released artifacts when available?

I don't know, but if not, maybe an option can be added?

Reply:
> Right. So my question is: can we use `metaFetch` to download the released artifacts when available?

Sure, feel free to amend the `default-fetcher` here:
https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/coq/meta-fetch/default.nix
You can add an extra "artifact" argument (true by default) to change the url in the github case here:
https://github.com/NixOS/nixpkgs/blob/a11ec092e86de8c887fe040ef9504562f31050c1/pkgs/build-support/coq/meta-fetch/default.nix#L38-L39

Reply:
> (false by default?)

Or maybe that default to the location.artifact attrset (and false there by default), so that we can control this repo by repo rather than release per release?

Reply:
I will propose a fix

Reply:
Many thanks.

Reply:
It seems I cannot push, the commit is here: https://github.com/NixOS/nixpkgs/commit/3fe6100bbc23130fb387c4f46635c18ac28ffc83

Reply:
I screwed up something, wait a sec...

Reply:
The hashes you put do not match mine :thinking: 
```raw
nix-build -A coqPackages_8_20.coq.ocamlPackages.elpi                                        (ocaml-elpi-clean|✚1)
these 2 derivations will be built:
  /nix/store/rfr5ih6dps8qxhld5ybczg1n41dhvg62-source.drv
  /nix/store/m7nvgzdpy70cw69kl83p4ys6hjvvpx39-ocaml4.14.2-elpi-1.18.1.drv
building '/nix/store/rfr5ih6dps8qxhld5ybczg1n41dhvg62-source.drv'...

trying https://github.com/LPCIC/elpi/releases/download/v1.18.1/elpi-1.18.1.tbz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 1251k  100 1251k    0     0  1690k      0 --:--:-- --:--:-- --:--:-- 2296k
unpacking source archive /build/elpi-1.18.1.tbz
error: hash mismatch in fixed-output derivation '/nix/store/rfr5ih6dps8qxhld5ybczg1n41dhvg62-source.drv':
         specified: sha256-rrIv/mVC0Ez3nU7fpnzwduIC3tI6l73DjgAbv1gd2v0=
            got:    sha256-NgRCFgzxOM7MCpMdnwxY/yYcoA8WKo+ZJrKJy1bWo0U=
error: 1 dependencies of derivation '/nix/store/m7nvgzdpy70cw69kl83p4ys6hjvvpx39-ocaml4.14.2-elpi-1.18
```

Reply:
Still my patch worked... @vbgl what am I doing wrong?

Reply:
I use `fetchurl` when you probably use `fetchzip`?

Reply:
Ah ... sure, but I thought the uncompressed archives were compared... why is the result different?

Reply:
`fetchurl` does not decompress: it computes the hash of the downloaded file.

Reply:
done: da329e41f978

Reply:
Thanks. Cherry-picked to this branch.